src/language.rs

```diff
@@ -44,11 +58,67 @@ impl SupportedLanguage for SupportedLanguageTypescript {
         tree_sitter_typescript::language_tsx()
     }
 
+    fn name(&self) -> SupportedLanguageName {
+        SupportedLanguageName::Typescript
+    }
+
     fn name_for_ignore_select(&self) -> &'static str {
         "ts"
     }
+
+    fn extensions(&self) -> Vec<&'static str> {
+        vec!["ts", "tsx"]
+    }
 }
 
 pub fn get_typescript_language() -> SupportedLanguageTypescript {
     SupportedLanguageTypescript
 }
+
+pub struct SupportedLanguageJavascript;
+
+impl SupportedLanguage for SupportedLanguageJavascript {
+    fn language(&self) -> Language {
+        tree_sitter_javascript::language()
+    }
+
+    fn name(&self) -> SupportedLanguageName {
+        SupportedLanguageName::Javascript
+    }
+
+    fn name_for_ignore_select(&self) -> &'static str {
+        "js"
+    }
+
+    fn extensions(&self) -> Vec<&'static str> {
+        vec!["js", "jsx", "vue", "cjs", "mjs"]
+    }
+}
+
+pub fn get_javascript_language() -> SupportedLanguageJavascript {
+    SupportedLanguageJavascript
+}
+
+pub fn get_all_supported_languages() -> HashMap<SupportedLanguageName, Box<dyn SupportedLanguage>> {
+    HashMap::from_iter([
+        (
+            SupportedLanguageName::Rust,
+            Box::new(get_rust_language()) as Box<dyn SupportedLanguage>,
+        ),
+        (
+            SupportedLanguageName::Typescript,
+            Box::new(get_typescript_language()) as Box<dyn SupportedLanguage>,
+        ),
+        (
+            SupportedLanguageName::Javascript,
+            Box::new(get_javascript_language()) as Box<dyn SupportedLanguage>,
+        ),
+    ])
+}
+
+pub fn maybe_supported_language_from_path(path: &Path) -> Option<Box<dyn SupportedLanguage>> {
```

Comment:
This is the core thing mapping project file path back to supported language, so this is getting called once per project file